pkg/apis/pingcap/v1alpha1/validation/validation.go

```diff
@@ -644,6 +644,8 @@ func validateUpdatePDConfig(oldPdSpec, pdSpec *v1alpha1.PDSpec, path *field.Path
 	return allErrs
 }
 
+// disallowMutateBootstrapSQLConfigMapName checks if user mutate the bootstrapSQLConfigMapName field.
+// Only allow to update bootstrapSQLConfigMapName from non-nil to nil.
```

Comment:
why do we need to support updates from non-nil to nil?

Reply:
If the user deletes the bootstrap sql configmap and also want to remove it from tidb pod.